### DISCUSSION TITLE
Igx-grid save default value on selecting other cells

### DISCUSSION THREAD

**Discussion by kevalgw** [IgniteUI/igniteui-angular#12248] [Q&A] [answered]
Hi, 

In my grid I want to save default value when user edit a particular cell and try to save null value
Please refer attached video for more reference

https://user-images.githubusercontent.com/101707048/198939851-d9e0d03d-71f7-40d5-82c3-7f16ae0e6db5.mp4

**Comment by ddaribo** [comment, marked as answer]
Hello @kevalgw,

While there is no built-in property in the grid for a cell editing default value, your requirement could easily be achieved with the methods exposed by the [IgxGridComponent API](https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/classes/IgxGridComponent.html).

The [cellEdit event](https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/classes/IgxGridComponent.html#cellEdit) of the grid could be handled in order to check whether the new value of the cell, which could be accessed from the event arguments’ `newValue` property, is equal to a null value for the corresponding data type. For example, for a ‘string’ that would be `‘’`, for a number – `0`, for a Date – `null`, etc.
Then, the specific default value for the particular column has to be set to the currently edited cell’s `editValue` and the `event.newValue` properties.
The cell could be retrieved with the help of the [getCellByKey](https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/classes/IgxGridComponent.html#getCellByKey) method of the grid.
How the default values are determined would depend on your own application logic. In [this sample app](https://stackblitz.com/edit/igx-grid-set-default-cell-values-when-editing-with-empty?file=src/app/grid/grid-editing-excel-style/grid-editing-excel-style.component.ts) I prepared for you, they are stored in a Map, for instance. Please, check it out.
